app/bot/openai.go

```diff
@@ -155,21 +176,57 @@ func (o *OpenAI) chatGPTRequest(request, userPrompt, sysPrompt string) (response
 		r = r[:maxMsgLen]
 	}
 
+	return o.chatGPTRequestInternal([]openai.ChatCompletionMessage{
+		{
+			Role:    openai.ChatMessageRoleSystem,
+			Content: sysPrompt,
+		},
+		{
+			Role:    openai.ChatMessageRoleUser,
+			Content: r,
+		},
+	})
+}
+
+func (o *OpenAI) shouldAnswerWithHistory(msg Message) bool {
+	if o.history.count < o.history.limit {
+		return false
+	}
+
+	if msg.Text[len(msg.Text)-1:] != "?" {
+		return false
+	}
+
+	// 10% chance to answer with ChatGPT for question
+	return o.rand(10) == 1
+}
+
+func (o *OpenAI) chatGPTRequestWithHistory(sysPrompt string) (response string, err error) {
```

Comment:
I'm not 100% sure, but i think we should reset history after this func done. This will eliminate random continuation of the same conversation. Generally speaking, I don't have strong opinion here, up to you

Reply:
I'd like to keep story for context answers if some questions were asked close.

Let's see how it works in practice